osf/models/provider.py

```diff
@@ -122,6 +122,7 @@ def update_or_create_from_json(cls, provider_data, user):
     primary_collection = models.ForeignKey('Collection', related_name='+',
                                            null=True, blank=True, on_delete=models.SET_NULL)
     name = models.CharField(null=False, max_length=128)  # max length on prod: 22
+    advertiseOnDiscoverPage = models.BooleanField(default=True, help_text="Indicates if the provider should be advertised on the discovery page.")
```

Comment:
Use underscores and no caps for `advertiseOnDiscoverPage` here please, it's ok in the HTML, but not on the backend. It should follow the convention of the other attribute names.